Raven.Database/Indexing/Index.cs

```diff
@@ -1070,7 +1070,20 @@ public static string TryGetDocKey(object current)
         {
             var dic = current as DynamicJsonObject;
             if (dic == null)
-                return null;
+            {                
+                var properites = current.GetType().GetProperties();
+                var idProperty = properites.FirstOrDefault(x => x.Name == "__document_id");
```

Comment:
1) Minor: please use `Constants.DocumentIdFieldName`
2) Should we check for `Constants.ReduceKeyFieldName` as well? (see below code)

Reply:
Done